test_bgp_default_originate.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2022 Nathan Mangar
+
+"""
+Test if default-originate works without route-map.
+"""
+
+__topotests_file__ = "bgp_default_route/test_bgp_default-originate.py"
+__topotests_gitrev__ = "4953ca977f3a5de8109ee6353ad07f816ca1774c"
+
+# pylint: disable=invalid-name, missing-class-docstring, missing-function-docstring, line-too-long, consider-using-f-string, wildcard-import, unused-wildcard-import, f-string-without-interpolation
+
+from topotato import *
+
+
+@topology_fixture()
+def topology(topo):
+    """
+    [ r1 ]
+      |
+    { s1 }
+      |
+    [ r2 ]
+    """
+    topo.router("r1").lo_ip4.append("172.16.255.254/32")
+    topo.router("r1").iface_to("s1").ip4.append("192.168.255.1/24")
+    topo.router("r2").iface_to("s1").ip4.append("192.168.255.2/24")
+
+
+class Configs(FRRConfigs):
+    routers = ["r1", "r2"]
+
+    zebra = """
+    #% extends "boilerplate.conf"
+    #% block main
+    #%   if router.name == 'r1'
+    interface lo
+     ip address {{ routers.r1.lo_ip4[0] }} 
+    !
+    #%   endif
+    #%   for iface in router.ifaces
+    interface {{ iface.ifname }}
+     ip address {{ iface.ip4[0] }} 
+    !
+    #%   endfor
+    ip forwarding
+    !
+    #% endblock
+    """
+
+    bgpd = """
+  #% block main
+    #%   if router.name == 'r2'
+    router bgp 65001
+      no bgp ebgp-requires-policy
+      neighbor {{ routers.r1.ifaces[0].ip4[0].ip }} remote-as 65000
+      neighbor {{ routers.r1.ifaces[0].ip4[0].ip }} timers 3 10
+      exit-address-family
+    !
+    #%   elif router.name == 'r1'
+    router bgp 65000
+      no bgp ebgp-requires-policy
+      neighbor {{ routers.r2.ifaces[0].ip4[0].ip }} remote-as 65001
+      neighbor {{ routers.r2.ifaces[0].ip4[0].ip }} timers 3 10
+      address-family ipv4 unicast
+        neighbor 192.168.255.2 default-originate
```

Comment:
Please don't mix explicit IP addresses `192.168.255.2` with template lookups `{{ routers.r2.ifaces[0].ip4[0].ip }}` above

(Ideally, should use lookups everywhere and remove hardcoded addresses where possible)

Reply:
Noted, will change that.

Reply:
You said you'd fix it but it's still there :)

Reply:
This is still here :D